Library/Homebrew/test/dev-cmd/test_spec.rb

```diff
@@ -17,4 +17,40 @@
       .and not_to_output.to_stderr
       .and be_a_success
   end
+
+  describe "using inreplace" do
```

Comment:
We usually try to limit the number of integration tests if possible. Is there any way to test this without integration tests?

Reply:
Probably! How does one get a generic formula instance in a unit test? (`inreplace` could be a static method, it doesn't use `Formula` state).

(Though I do think we should have more integration tests using formula `test` blocks, such as the one used to repro this issue: https://github.com/Homebrew/brew/pull/15807#issuecomment-1664172239 )

Reply:
Ideally, we'd add some tests like those found in the `formula_spec.rb` file. Here's an example with the livecheck block.

https://github.com/Homebrew/brew/blob/3eb48a828d44643b8d427f647227a277b1a4c1bf/Library/Homebrew/test/formula_spec.rb#L665-L679

If that's too much of an ask, maybe we could just combine these integration tests. I think that would still result in less CI time.

Reply:
We don't seem to test the test blocks that much though.

Reply:
Agreed these would ideally not be integration tests.